### ISSUE TITLE
Difference Between TransactionMut::after_state and ReadTxn::state_vector

### ISSUE THREAD

**Issue by maldrake** [y-crdt/y-crdt#251]
Hi,

When editing a document, I am seeing a discrepancy in the state vectors returned by different methods.  A minimum reproduction of the scenario is as follows:

```
    #[test]
    fn after_state_v_state_vector() {
        let doc = Doc::new();

        let text = doc.get_or_insert_text("data");
        let mut txn = doc.transact_mut();
        text.push(&mut txn, "Test string");
        txn.commit();

        let sv1 = txn.after_state();
        let sv2 = txn.state_vector();

        assert_eq!(sv1, &sv2);
    }
```

When I run the above code, the test case fails as follows:

```
test tests::after_state_v_state_vector ... thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `StateVector({})`,
 right: `StateVector({2634145553: 11})`', src/lib.rs:200:9
```

I had expected that both methods of obtaining a state vector -- `txn.after_state()` and `txn.state_vector()` would return a state vector including the edit made to insert "Test string."  It appears instead that the state vector returned by `after_state` is empty.  Is this a bug or a misunderstanding of the API on my part?

**Comment by Horusiath** [comment]
It's a documentation issue on our side: `after_state` describes document's state after the transaction has been committed. Current state can be tracked using `state_vector`. #250 will address that.

**Comment by maldrake** [comment]
Wouldn't the two be equivalent after the transaction is committed, as in the test case above?

**Comment by Horusiath** [comment]
This part is due to bug, that has been already fixed and merged into main branch ;)

**Comment by maldrake** [comment]
Oh, nice!

**Comment by Horusiath** [comment]
Changes have been included in yrs release v0.16. I'm closing this now. Feel free to reopen if you still see the problems.
